packages/material-ui/src/Popper/Popper.test.js

```diff
@@ -169,6 +167,32 @@ describe('<Popper />', () => {
     });
   });
 
+  describe('prop: keepMounted', () => {
+    it('should keep the children in the DOM', () => {
+      const wrapper = mount(<Popper {...defaultProps} open={false} keepMounted />);
+      assert.strictEqual(wrapper.find('span').length, 1);
+    });
+
+    /* Test case for https://github.com/mui-org/material-ui/issues/15180 */
+    it('should remove from DOM when closed whilst transition has entering status', () => {
+      const wrapper = mount(
+        <Popper {...defaultProps} open={false} keepMounted={false} transition>
+          {({ TransitionProps }) => (
+            <Grow {...TransitionProps}>
+              <span>Hello World</span>
+            </Grow>
+          )}
+        </Popper>,
+      );
+      wrapper.setProps({ open: true });
+      wrapper.update();
+      assert.strictEqual(wrapper.find('span').length, 1);
```

Comment:
I think we should use a proper selector here. Probably best add an id to the child and look for that one. Otherwise changing the DOM tree would break this test.

Reply:
I've updated to using an id selector, and tried to make the id as "unique" as possible to ensure it doesn't accidentally clash with anything placed into the `Popper` in the future. Have also updated the other Popper tests to use the id, and made sure they all pass as expected.